sql/updates/latest-dev.sql

```diff
@@ -397,3 +397,33 @@ CREATE FUNCTION _timescaledb_functions.constraint_clone(constraint_oid OID,targe
 DROP FUNCTION IF EXISTS _timescaledb_functions.chunks_in;
 DROP FUNCTION IF EXISTS _timescaledb_internal.chunks_in;
 
+-- Remove unwanted entries from extconfig and extcondition in pg_extension
+DO $block$
+DECLARE r record;
+position integer;
+value varchar;
+BEGIN
+    FOR r IN SELECT oid FROM pg_class WHERE oid in
+        (
+         '_timescaledb_cache.cache_inval_hypertable'::regclass,
+         '_timescaledb_cache.cache_inval_extension'::regclass,
+         '_timescaledb_cache.cache_inval_bgw_job'::regclass,
+         '_timescaledb_internal.job_errors'::regclass
+        )
+    LOOP
+        -- Get the index position for this function. We need to remove entries from both
+        -- extcondition and extconfig at this position.
+        SELECT array_position(extconfig, r.oid) INTO position FROM pg_extension
+            WHERE extname = 'timescaledb';
+        -- This should not happen if this block is called once, but better be safe
+        IF position IS NULL THEN
+            CONTINUE;
+        END IF;
+        -- We cannot use array_remove here since it will remove multiple matching entries
+        -- unnecessarily. So, we concatenate sub arrays excluding the index entry
+        UPDATE pg_extension SET
+            extconfig = extconfig[:position - 1] || extconfig[position + 1:],
+            extcondition = extcondition[:position - 1] || extcondition[position + 1:]
+            WHERE extname = 'timescaledb';
+   END LOOP;
+END $block$ ;
```

Comment:
Why don't use `ALTER EXTENSION timescaledb DROP TABLE ...` for the given objects?

Reply:
@fabriziomello while it does remove entries for extconfig and extcondition, it also completely disassociates the table from the extension. So, it's not useful at all 👎 

```
postgres=# create extension timescaledb;
CREATE EXTENSION
postgres=# ALTER EXTENSION timescaledb DROP TABLE  _timescaledb_config.bgw_job;
ALTER EXTENSION
postgres=# drop extension timescaledb ;
ERROR:  cannot drop extension timescaledb because other objects depend on it
DETAIL:  table _timescaledb_config.bgw_job depends on schema _timescaledb_config
HINT:  Use DROP ... CASCADE to drop the dependent objects too.
```

Reply:
Good point, but in any case you can do an `ALTER EXTENSION timescaledb DROP` followed by an `ALTER EXTENSION timescaledb ADD` to disassociate and associate again the object without the dump configuration.